storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java

```
@@ -0,0 +1,570 @@
+
+        @Override
+        public void validateField(String name, Object o) {
+            if (o == null) {
+                return;
+            }
+            ConfigValidationUtils.NestableFieldValidator validator = ConfigValidationUtils.mapFv(ConfigValidationUtils.fv(String.class, false),
+                    ConfigValidationUtils.mapFv(ConfigValidationUtils.fv(String.class, false),
+                            ConfigValidationUtils.listFv(String.class, false), false), true);
+            validator.validateField(name, o);
+        }
+    }
+
+    /**
+     * validates a list of has no duplicates
+     */
+    public static class NoDuplicateInListValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object field) {
+            if (field == null) {
+                return;
+            }
+            //check if iterable
+            SimpleTypeValidator.validateField(name, Iterable.class, field);
+            HashSet<Object> objectSet = new HashSet<Object>();
+            for (Object o : (Iterable) field) {
+                if (objectSet.contains(o)) {
+                    throw new IllegalArgumentException(name + " should contain no duplicate elements. Duplicated element: " + o);
+                }
+                objectSet.add(o);
+            }
+        }
+    }
+
+    /**
+     * Validates a String or a list of Strings
+     */
+    public static class StringOrStringListValidator extends Validator {
+
+        private ConfigValidationUtils.FieldValidator fv = ConfigValidationUtils.listFv(String.class, false);
+
+        @Override
+        public void validateField(String name, Object o) {
+
+            if (o == null) {
+                return;
+            }
+            if (o instanceof String) {
+                return;
+            }
+            //check if iterable
+            try {
+                SimpleTypeValidator.validateField(name, Iterable.class, o);
+            } catch (Exception ex) {
```

Why are we ignoring the exception?